packages/google_maps_flutter/google_maps_flutter/example/test_driver/google_maps_e2e.dart

```diff
@@ -341,6 +341,46 @@ void main() {
     expect(scrollGesturesEnabled, true);
   });
 
+  testWidgets('testInitialCenterLocationAtCenter', (WidgetTester tester) async {
+    final Completer<GoogleMapController> mapControllerCompleter =
+        Completer<GoogleMapController>();
+    final Key key = GlobalKey();
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: GoogleMap(
+          key: key,
+          initialCameraPosition: _kInitialCameraPosition,
+          onMapCreated: (GoogleMapController controller) {
+            mapControllerCompleter.complete(controller);
+          },
+        ),
+      ),
+    );
+    final GoogleMapController mapController =
+        await mapControllerCompleter.future;
+
+    // `onMapCreated` is called before the map is actually ready on the platform thread.
+    // Wait for a second until the map is settled down.
+    await tester.pumpAndSettle();
+    await Future.delayed(Duration(seconds: 1));
```

Comment:
Do we have a reliable way to detect when it's ready?

Reply:
Unless we add a new callback API in the google map, something like "onMapRendered". This can be in a separate PR tho.

Reply:
We should do that, I'm not blocking this PR since we're trading a 3 seconds timer with a 1 second one , but we should at least add a TODO.

Reply:
Will do!